modules/task/backend.go

```diff
@@ -114,22 +113,13 @@ func (b *inMemBackend) Start(ready chan<- struct{}) <-chan error {
 }
 
 func (b *inMemBackend) consumeFromQueue(errorCh chan error) {
-	for {
-		select {
-		case msg, ok := <-b.bufQueue:
-			if ok {
-				errorCh <- Run(msg)
-			} else {
-				return
-			}
-		case <-time.After(time.Millisecond):
-			ulog.Logger().Error("Timed out after 1 ms")
-		}
+	for msg := range b.bufQueue {
+		errorCh <- Run(context.Background(), msg)
```

Comment:
This is ok for now, but we need to have a common serialization between backends that don't support context propagation.